vllm/logger.py

```diff
@@ -10,31 +10,39 @@
 from logging.config import dictConfig
 from os import path
 from types import MethodType
-from typing import Any, Optional, cast
+from typing import Any, Dict, Optional, cast
 
 import vllm.envs as envs
 
 VLLM_CONFIGURE_LOGGING = envs.VLLM_CONFIGURE_LOGGING
 VLLM_LOGGING_CONFIG_PATH = envs.VLLM_LOGGING_CONFIG_PATH
 VLLM_LOGGING_LEVEL = envs.VLLM_LOGGING_LEVEL
 VLLM_LOGGING_PREFIX = envs.VLLM_LOGGING_PREFIX
+VLLM_LOG_FILTER_PATTERNS = envs.VLLM_LOG_FILTER_PATTERNS
 
 _FORMAT = (f"{VLLM_LOGGING_PREFIX}%(levelname)s %(asctime)s "
            "%(filename)s:%(lineno)d] %(message)s")
 _DATE_FORMAT = "%m-%d %H:%M:%S"
 
-DEFAULT_LOGGING_CONFIG = {
+DEFAULT_LOGGING_CONFIG: Dict[str, Any] = {
     "formatters": {
         "vllm": {
             "class": "vllm.logging_utils.NewLineFormatter",
             "datefmt": _DATE_FORMAT,
             "format": _FORMAT,
         },
     },
+    "filters": {
+        "vllm_redact": {
+            "()": "vllm.logging_utils.RedactFilter",
```

Comment:
Can you explain a bit about `()` vs `class` here?

From a quick google I'm not exactly sure when to use one or the other

Reply:
I assumed I just needed to use `class` too, but I found https://docs.python.org/3/howto/logging-cookbook.html#configuring-filters-with-dictconfig explaining that if you define your own `Filter` subclass, you use `()` to specify the callable for the filter.

Reply:
ohhh TIL about the logging cookbook. Thanks!
(Might be worth popping that link in as a comment here too)